src/containers/WorkspaceView.tsx

```diff
@@ -1,50 +1,184 @@
+}
+
+type WorkspaceRecordingsRoute = {
+    page: 'recordings',
+    workspaceName: string
+}
+type WorspaceRecordingRoute = {
+    page: 'recording',
+    recordingId: string,
+    workspaceName: string
+}
+type WorspaceSortingRoute = {
+    page: 'sorting',
+    recordingId: string,
+    sortingId: string,
+    workspaceName: string
+}
+type WorkspaceRoute = WorkspaceRecordingsRoute | WorspaceRecordingRoute | WorspaceSortingRoute
+type GotoRecordingsPageAction = {
+    type: 'gotoRecordingsPage'
+}
+type GotoRecordingPageAction = {
+    type: 'gotoRecordingPage',
+    recordingId: string
+}
+type GotoSortingPageAction = {
+    type: 'gotoSortingPage',
+    recordingId: string,
+    sortingId: string
+}
+type WorkspaceRouteAction = GotoRecordingsPageAction | GotoRecordingPageAction | GotoSortingPageAction
+export type WorkspaceRouteDispatch = (a: WorkspaceRouteAction) => void
+
+const routeFromLocation = (location: {pathname: string, search: string}): WorkspaceRoute => {
+    const pathList = location.pathname.split('/')
+    const workspaceName = pathList[1] || 'default'
+    const page = pathList[2] || 'recordings'
+    if (!isPage(page)) throw Error(`Invalid page: ${page}`)
+    switch (page) {
+        case 'recordings': return {
+            workspaceName,
+            page
+        }
+        case 'recording': return {
+            workspaceName,
+            page,
+            recordingId: pathList[3]
+        }
+        case 'sorting': return {
+            workspaceName,
+            page,
+            recordingId: pathList[3] || '',
+            sortingId: pathList[4] || ''
+        }
+        default: return {
+            workspaceName,
+            page: 'recordings'
+        }
+    }
+}
+
+var queryString = (params: {[key: string]: string}) => {
+    const keys = Object.keys(params)
+    if (keys.length === 0) return ''
+    return '?' + (
+        keys.map((key) => {
+            return encodeURIComponent(key) + '=' + encodeURIComponent(params[key])
+        }).join('&')
     )
 }
 
+const locationFromRoute = (route: WorkspaceRoute, workspaceInfo: WorkspaceInfo) => {
+    const queryParams: {[key: string]: string} = {}
+    if (workspaceInfo.feedUri) {
+        queryParams['feed'] = workspaceInfo.feedUri
+    }
+    switch(route.page) {
+        case 'recordings': return {
+            pathname: `/${route.workspaceName}`,
+            search: queryString(queryParams)
+        }
+        case 'recording': return {
+            pathname: `/${route.workspaceName}/recording/${route.recordingId}`,
+            search: queryString(queryParams)
+        }
+        case 'sorting': return {
+            pathname: `/${route.workspaceName}/sorting/${route.recordingId}/${route.sortingId}`,
+            search: queryString(queryParams)
+        }
+    }
+}
+
+const WorkspaceView: FunctionComponent<Props> = ({ width, height, sortings, recordings, onDeleteRecordings, workspaceInfo, plugins }) => {
+    const history = useHistory()
```

Comment:
This is the react-router history object that allows us to push locations to the history